model/job-management/src/main/proto/beam_job_api.proto

```diff
@@ -151,8 +152,9 @@ message GetJobStateRequest {
 
 }
 
-message GetJobStateResponse {
+message JobStateEvent {
   JobState.Enum state = 1; // (required)
+  google.protobuf.Timestamp timestamp = 2; // (required)
```

Comment:
Why is the timestamp not set a the receiving side? Do we need to include this in the Proto?

Reply:
One of the main users stories behind this PR is that there could be multiple clients, and some may join the stream late, so those late joining clients won't know the timestamps.